src/Core/Mesh/TriangleMesh.hpp

```diff
@@ -19,67 +17,83 @@ namespace Core
 /// The VertexAttribManager allows to dynammicaly add VertexAttrib per Vertex
 /// See MeshUtils for geometric functions operating on a mesh.
 /// Points and Normals are always present, accessible with
-/// points() and normals()
+/// points() and normals().
 /// Other attribs could be added with attribManager().addAttrib() and
 /// accesssed with attribManager().getAttrib()
-class TriangleMesh
-{
+/// Note: attribute names "in_position" and "in_normal" are reserved.
+class TriangleMesh {
   public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
-    using PointAttribHandle  = AttribHandle<Vector3>;
+    using PointAttribHandle = AttribHandle<Vector3>;
     using NormalAttribHandle = AttribHandle<Vector3>;
-    using FloatAttribHandle  = AttribHandle<float>;
-    using Vec2AttribHandle   = AttribHandle<Vector2>;
-    using Vec3AttribHandle   = AttribHandle<Vector3>;
-    using Vec4AttribHandle   = AttribHandle<Vector4>;
-    using Face               = VectorNui;
+    using FloatAttribHandle = AttribHandle<float>;
+    using Vec2AttribHandle = AttribHandle<Vector2>;
+    using Vec3AttribHandle = AttribHandle<Vector3>;
+    using Vec4AttribHandle = AttribHandle<Vector4>;
+    using Face = VectorNui;
 
     /// Create an empty mesh.
     inline TriangleMesh() { initDefaultAttribs(); }
-    /// Copy constructor and assignment operator
-    /// \warning No deep copy of attributes !
+
+    /// Copy constructor and assignment operator.
+    /// Note: shallow copy of attributes.
```

Comment:
From my point of view, keeping the shallow copy by default is misleading. Do you have a specific need for that ?

I see that could be useful to perform partial copies, in that case we need to have a mechanism for that.

Reply:
Well, I had a discussion with @dlyr about that point.
When doing a deep copy of a mesh, you may need only some attributes you need / are aware of, not all. I don't really remember all the discussion though, but I ended up restoring shallow copy as the default behavior.

Reply:
So we wait @dlyr is back before taking a decision.

Reply:
Actually, the shallow copy is highly appreciated as the default behavior for functions/methods returning a `TriangleMesh`.
Thus I left it as is, renamed `copyAttribs` into `partialCopy` and added a `fullCopy` method.